### PR TITLE
Rewritten docstring for index_by in tables.py

### PR DESCRIPTION
**Pull request by farhan-naqvi**
Rewritten index_by function 's docstring(from line 2901) in detail and mentioned 2 examples along with their output.

### CODE REVIEW

File: datascience/tables.py [hunk 1]
```diff
@@ -1,5 +1,12 @@
 """Tables are sequences of labeled columns."""
 
+import datascience.predicates as _predicates
```
Comment (davidwagner) [inline datascience/tables.py:3]: Please avoid making unnecessary changes, like re-ordering imports.  We would prefer to see a clean diff that only changes the parts that you modified.

File: datascience/tables.py [hunk 2]
```diff
@@ -16,22 +23,16 @@
...
-import datascience.util as _util
-from datascience.util import make_array
-import datascience.predicates as _predicates
 
 # intitializing go and make_subplots as globals set to None
 go, make_subplots = None, None
 
 _INTERACTIVE_PLOTS = False
 
+
```
Comment (davidwagner) [inline datascience/tables.py:35]: Please avoid unnecessary diffs.

File: datascience/tables.py [hunk 3]
```diff
@@ -79,7 +80,8 @@ def empty(cls, labels=None):
         Returns:
             A new instance of ``Table``.
         """
-        warnings.warn("Table.empty(labels) is deprecated. Use Table(labels)", FutureWarning)
+        warnings.warn(
```
Comment (davidwagner) [inline datascience/tables.py:83]: Please don't add extra whitespace - this makes this actively worse.

File: datascience/tables.py [hunk 4]
```diff
@@ -2882,8 +2899,53 @@ def as_html(self, max_rows=0):
         return '\n'.join(4 * indent * ' ' + text for indent, text in lines)
 
     def index_by(self, column_or_label):
-        """Return a dict keyed by values in a column that contains lists of
-        rows corresponding to each value.
+        """
+        Returns a dictionary keyed by values in a column that contains lists of rows corresponding to each value.
+
+        Args:
+            self:
```
Comment (davidwagner) [inline datascience/tables.py:2906]: No need to document the meaning of `self`

File: datascience/tables.py [hunk 5]
```diff
@@ -2882,8 +2899,53 @@ def as_html(self, max_rows=0):
...
+        Returns a dictionary keyed by values in a column that contains lists of rows corresponding to each value.
+
+        Args:
+            self: 
+
+                Table on which the function is to be applied.
+
+            column_or_label: 
+
+                Column whose values will be considered as keys in the dictionary that will be returned.(Only one columns can be passed)
```
Comment (davidwagner) [inline datascience/tables.py:2912]: "one column"

**Comment by davidwagner** [comment]
Thanks for preparing this docstring!  This looks like it would be useful to incorporate.

However, I can't accept this particular pull request.  The diff shows many extraneous changes, e.g., whitespace changes throughout.  This makes it hard to review, clutters up the history, and risks creating merge conflicts in the future.  Can you create a new pull request that has a "clean" diff, that changes only the specific part of the code that you are modifying/adding, and leaves everything else unchanged?  Thank you.

**Comment by adnanhemani** [comment]
Closing this PR as it contains too many unnecessary changes and there hasn't been any activity on this PR recently. Please feel free to raise a new PR if you decide to work on this again and incorporate the changes requested in your new work. Thanks for your efforts!
